File: docs/src/pages/GuidePage/content/calculation/code/calculate_assembly.py
from lcax import calculate_assembly, CalculationOptions, LifeCycleModule, ImpactCategoryKey, Assembly
import json

with open("assembly.json") as f:
    assembly = Assembly(**json.load(f))

options = CalculationOptions(
    reference_study_period=50,
    life_cycle_modules=[LifeCycleModule.A1A3, LifeCycleModule.A4],
    impact_categories=[ImpactCategoryKey.Gwp],
    overwrite_existing_results=True
)

impacts = calculate_assembly(assembly, options)

File: docs/src/pages/GuidePage/content/calculation/code/calculateAssembly.ts
import { calculateAssembly, CalculationOptions, LifeCycleModule, ImpactCategoryKey } from 'lcax'
import fs from 'fs'

const assembly_data = fs.readFileSync('assembly.json')
const assembly = JSON.parse(assembly_data)

const options: CalculationOptions = {
  referenceStudyPeriod: 50,
  lifeCycleModules: ['a1a3', 'a4'] as LifeCycleModule[],
  impactCategories: ['gwp'] as ImpactCategoryKey[],
  overwriteExistingResults: true,
}

const impacts = calculateAssembly(assembly, options)
